dbft.go

```diff
@@ -220,8 +220,8 @@ func (d *DBFT) OnReceive(msg payload.ConsensusPayload) {
 	if msg.Height() < d.BlockIndex {
 		d.Logger.Debug("ignoring old height", zap.Uint32("height", msg.Height()))
 		return
-	} else if msg.Height() > d.BlockIndex || (msg.Height() == d.BlockIndex &&
-		msg.ViewNumber() == d.ViewNumber+1 && msg.Type() != payload.RecoveryMessageType) {
+	} else if msg.Height() > d.BlockIndex ||
+		(msg.ViewNumber() > d.ViewNumber && msg.Type() != payload.RecoveryMessageType) {
```

Comment:
Agree with upper views, but why message height check is removed? Do we really need messages from lower heights?

Reply:
Maybe it's somewhat misleading, but if we're here, then `msg.Height() == d.BlockIndex` already.